shell/common/shell.cc

```diff
@@ -1240,4 +1251,17 @@ Rasterizer::Screenshot Shell::Screenshot(
   return screenshot;
 }
 
+bool Shell::WaitForFirstFrame(fml::TimeDelta timeout) {
+  FML_DCHECK(is_setup_);
+  FML_DCHECK(!task_runners_.GetUITaskRunner()->RunsTasksOnCurrentThread());
```

Comment:
Can we ensure that this only gets called on the platform task runner?

Reply:
That is a bit fragile and not a technical requirement, it may interfere with non-standard threading models.  I've switched up the checks as part of your your other feedback.